test/NameRegistry.t.sol

```diff
@@ -1470,12 +1468,40 @@ contract NameRegistryTest is Test {
         nameRegistry.ownerOf(ALICE_TOKEN_ID);
     }
 
+    function testFuzzCannotSetApproverIfNotOwner(
```

Comment:
this test can be simplified quite a bit - i don't think you need to have recovery or many of the assertions, you can do something like: 


```suggestion
    function testFuzzCannotSetApproverIfNotOwner(
        address alice,
        address bob,
        address approver
    ) public {
        _assumeClean(alice);
        _assumeClean(recovery);
        _assumeClean(approver);
        vm.assume(alice != bob && alice != approver);
        vm.assume(bob != address(0));
        _register(alice);

        // bob tries to set an approver for alices fname
        vm.prank(bob);
        vm.expectRevert("ERC721: approve caller is not token owner or approved for all");
        nameRegistry.approve(approver, ALICE_TOKEN_ID);

       // call ERC721 approver and assert that it is not bob one for alice's token id 
    }

```

Reply:
Awesome, yeah wasnt sure if I should use these redundant checks. They have been removed. Appreciate the guidance.